components/datetime/src/raw/datetime.rs

```diff
@@ -194,6 +194,14 @@ impl DateFormat {
             + ResourceProvider<WeekDataV1Marker>
             + ?Sized,
     {
+        let cal = locale.extensions.unicode.keywords.get(&key!("ca"));
+        if cal == Some(&value!("ethioaa")) {
+            locale
+                .extensions
+                .unicode
+                .keywords
+                .set(key!("ca"), value!("ethiopic"));
```

Comment:
Nit, here and elsewhere (3 places total): I'm not really happy about calendar-specific code in the icu_datetime crate and in a hot path like this. I don't have an alternative to suggest right now; would need to think more about it. Maybe file an issue?

Reply:
Correct my understanding if I'm wrong: for japanese/japanext, we duplicate the data and rely on deduplication to take care of it for us, but for ethiopic/ethioaa, we do this hard-coded transformation?

Seems like we should do the same solution in both cases

Reply:
Yes, because ethiopic/ethioaa data is assumed identical beyond the era data, and it amounts to one additional era in the data file. Whereas for Japanese we're explicitly trying to reduce data size for nondeduplicatable data. We could be uniform, that introduces the chance for more bugs from mixing calendars, though.

Reply:
Thought: Maybe the calendar trait should return a calendar ID for lookup of different types of calendar data. Ethiopian can return the same calendar ID for all keys, and japanese can return the same ID for date patterns but not date symbols, for example.